### DISCUSSION TITLE
TypeError: ctx.ellipse is not a function

### DISCUSSION THREAD

**Discussion by vikasacharya16** [iddan/react-native-canvas#268] [Q&A] [answered]
this is working
```javascript
ctx.fillRect(x - 50, y - 50, 100, 100);  
ctx.arc(x, y, 50, 0, Math.PI + (Math.PI * 2) / 2, 3 % 2 == 1);
```
this is not working 
```javascript
ctx.ellipse(x, y, 50, 50, Math.PI / 4, 0,  2 * Math.PI);
```
giving an error
```javascript
TypeError: ctx.ellipse is not a function. (In 'ctx.ellipse(x, y, 50, 50, Math.PI / 4, 0, 2 * Math.PI)', 'ctx.ellipse' is undefined)
```
I'm working on Adroid.

**Comment by michaelmika** [comment]
For me it is also needed to draw ellipses. Would be great if it would be supported in future.

For now, you can use a polyfill such as: [https://gist.github.com/floriancargoet/20cede87a76b4073177f](https://gist.github.com/floriancargoet/20cede87a76b4073177f)

**Comment by iddan** [comment, marked as answer]
Fixed in react-native-canvas@0.1.39
